src/arrayMethodSort.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let temp = 0;
+    let comparator = compareFunction;
+
+    if (arguments.length === 0) {
+      comparator = (x, y) => {
+        const stringX = String(x);
+        const stringY = String(y);
+
+        if (stringX > stringY) {
+          return 1;
+        } else if (stringX === stringY) {
+          return 0;
+        } else if (stringX < stringY) {
+          return -1;
+        }
```

Comment:
You can move this logic to the separate function
```suggestion
      comparator = (x, y) => {
        const stringX = String(x);
        const stringY = String(y);

        if (stringX > stringY) {
          return 1;
        }
        
        if (stringX < stringY) {
          return -1;
        }
        
        return 0
```
Also, there is no need for `else if` as the code will not go further after `return` anyway